### ISSUE TITLE
zed writing to pool disks (a lot) for 20+ minutes on reboot

### ISSUE THREAD

**Issue by DFloyd22** [openzfs/zfs#12977]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Ubuntu
Distribution Version	| 20.04.3 LTS
Kernel Version	| 5.4.0-94-generic  
Architecture	| Intel 64bit
OpenZFS Version	| zfs-0.8.3-1ubuntu12.13/zfs-kmod-0.8.3-1ubuntu12.13

### Describe the problem you're observing
Running `sudo iftop` shows that `zed` is writing significantly more than normal (constantly?) for 20+ minutes at a time. Running `top` and `ps aux` shows that `txg_sync` and multiple instances of z_null_iss are running (at least the `z_null_iss` instances stay running for 20+ mins, possibly the `txg_sync` too, I wasn’t paying attention to that one though).
#### Prerequisites:

1. A raidz2-0 pool with all disks online on an Ubuntu 20.04.3 LTS release.
2. The zpool status command reporting: “No known data errors”.
3. I'll include this, although it might be irrelevant: snapshots taken at 15 min intervals with sanoid (sanoid.conf contents are below):

#### Steps:
1. Run the command: sudo apt-get install --install-recommends linux-generic-hwe-20.04
2. Reboot to **use the new 5.11 kernel**, not the old 5.4 one - (in bold because I wasn't sure what kernel version to put in the system info at the top).

Sanoid.conf:
[qpool]
        use_template = production
        recursive = yes
        process_children_only = no
#############################
#templates below this line #
#############################
[template_production]
        frequently = 6
        hourly = 36
        daily = 30
        monthly = 12
        yearly = 0
        autosnap = yes
        autoprune = yes

### Include any warning/errors/backtraces from the system logs

From var/log/syslog
```
an 13 21:27:29 my-ubuntu-desktop systemd[1]: Started ZFS Event Daemon (zed).
Jan 13 21:27:29 my-ubuntu-desktop zed[2554]: ZFS Event Daemon 0.8.3-1ubuntu12.13 (PID 2554)
Jan 13 21:27:29 my-ubuntu-desktop zed[2554]: Processing events since eid=0
Jan 13 21:27:29 my-ubuntu-desktop zed: eid=1 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:29 my-ubuntu-desktop zed: eid=2 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:30 my-ubuntu-desktop zed: eid=3 class=pool_import pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:30 my-ubuntu-desktop zed: eid=4 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:30 my-ubuntu-desktop zed: eid=5 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:31 my-ubuntu-desktop zed: eid=6 class=vdev_autoexpand pool_guid=0xD3077FA19A75F9B3 vdev_path=/dev/disk/by-id/ata-SAMSUNG_HD154UI_Z1YWD3Z3221331-part1 vdev_state=ONLINE
Jan 13 21:27:31 my-ubuntu-desktop zed: eid=7 class=vdev_autoexpand pool_guid=0xD3077FA19A75F9B3 vdev_path=/dev/disk/by-id/ata-WDC_WD15EADS-00S2B0_WD-WDDYY014242131-part1 vdev_state=ONLINE
Jan 13 21:27:33 my-ubuntu-desktop zed: eid=8 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:33 my-ubuntu-desktop zed: eid=9 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:35 my-ubuntu-desktop zed: eid=10 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:35 my-ubuntu-desktop zed: eid=11 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:36 my-ubuntu-desktop zed: eid=12 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:36 my-ubuntu-desktop zed: eid=13 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:38 my-ubuntu-desktop zed: eid=14 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:27:38 my-ubuntu-desktop zed: eid=15 class=config_sync pool_guid=0xD3077FA19A75F9B3  
 
moving ahead 15 - 20 minutes ...

Jan 13 21:46:14 my-ubuntu-desktop zed: eid=1587 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:46:15 my-ubuntu-desktop zed: eid=1588 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:46:15 my-ubuntu-desktop zed: eid=1589 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:46:16 my-ubuntu-desktop zed: eid=1590 class=config_sync pool_guid=0xD3077FA19A75F9B3  
Jan 13 21:46:16 my-ubuntu-desktop zed: eid=1591 class=config_sync pool_guid=0xD3077FA19A75F9B3  

nothing more until the 22:00 snapshot

Jan 13 22:00:01 my-ubuntu-desktop sanoid[282146]: INFO: taking snapshots...
Jan 13 22:00:01 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool/compressed@autosnap_2022-01-13_22:00:01_hourly
Jan 13 22:00:01 my-ubuntu-desktop CRON[282389]: (root) CMD ([ -f /etc/sanoid/sanoid.conf ] && if [ ! -d /run/systemd/system 
]; then TZ=UTC /usr/sbin/sanoid --cron --quiet; fi)
Jan 13 22:00:01 my-ubuntu-desktop zed: eid=1592 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:01 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool/compressed@autosnap_2022-01-13_22:00:01_frequently
Jan 13 22:00:01 my-ubuntu-desktop zed: eid=1593 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:02 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool/games@autosnap_2022-01-13_22:00:01_hourly
Jan 13 22:00:02 my-ubuntu-desktop zed: eid=1594 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:02 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool/games@autosnap_2022-01-13_22:00:01_frequently
Jan 13 22:00:02 my-ubuntu-desktop zed: eid=1595 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:02 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool@autosnap_2022-01-13_22:00:01_hourly
Jan 13 22:00:02 my-ubuntu-desktop zed: eid=1596 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:03 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool@autosnap_2022-01-13_22:00:01_frequently
Jan 13 22:00:03 my-ubuntu-desktop zed: eid=1597 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:03 my-ubuntu-desktop sanoid[282146]: taking snapshot qpool/deduplicated@autosnap_2022-01-13_22:00:01_hourly

This repeats until...

Jan 13 22:00:09 my-ubuntu-desktop sanoid[283424]: INFO: pruning qpool/deduplicated@autosnap_2022-01-09_20:00:13_hourly ...
Jan 13 22:00:09 my-ubuntu-desktop zed: eid=1611 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:00:09 my-ubuntu-desktop sanoid[283424]: INFO: removing destroyed snapshots from cache.
Jan 13 22:00:09 my-ubuntu-desktop systemd[1]: sanoid-prune.service: Succeeded.

Nothing more until the next snapshot

Jan 13 22:15:01 my-ubuntu-desktop sanoid[298738]: INFO: taking snapshots...
Jan 13 22:15:01 my-ubuntu-desktop sanoid[298738]: taking snapshot qpool/compressed@autosnap_2022-01-13_22:15:01_frequently
Jan 13 22:15:01 my-ubuntu-desktop zed: eid=1612 class=history_event pool_guid=0xD3077FA19A75F9B3  
Jan 13 22:15:01 my-ubuntu-desktop CRON[298759]: (root) CMD ([ -f /etc/sanoid/sanoid.conf ] && if [ ! -d /run/systemd/system 
]; then TZ=UTC /usr/sbin/sanoid --cron --quiet; fi)

```

**Comment by AllKind** [comment]
https://github.com/openzfs/zfs/issues/10087#issuecomment-698630276

> Jan 13 21:27:31 my-ubuntu-desktop zed: eid=6 class=vdev_autoexpand pool_guid=0xD3077FA19A75F9B3 vdev_path=/dev/disk/by-id/ata-SAMSUNG_HD154UI_Z1YWD3Z3221331-part1 vdev_state=ONLINE

Is autoexpand enabled on the pool?
Try disabling it, to see if that is causing the behavior.

**Comment by DFloyd22** [comment]
It does have autoexpand enabled. Unfortunately I panicked, reverted back to the old kernel (5.4.0-94-generic) and apt removed 5.11 and the HWE stack. When I get a backup sorted out I can try 5.11 again a try some troubleshooting. 

The 5.4 kernel produced the same autoexpand message in syslog, but the zed messages stop at eid=79:
```
Jan 15 13:18:18 my-ubuntu-desktop zed: eid=79 class=history_event pool_guid=...
```

**Comment by AllKind** [comment]
autoexpand is a pool property.

**Comment by DFloyd22** [comment]
Yes I know, it's the property I accessed in order to make the "It does have autoexpand enabled" statement.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by behlendorf** [comment]
For completeness this should have been resolved by e27e692bcc2c3e5b79f60ef16a2183f2231ff012 which was backported for the 2.1.6 release.
